### PR TITLE
Feat parse catalog

### PR DESCRIPTION
**Pull request by jim-gyas**
Added Parse catalog implementation with test

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,14 +16,14 @@ classifiers = [
     "License :: OSI Approved :: MIT License",
     "Operating System :: OS Independent",
 ]
-
 dependencies = [
-    "git+https://github.com/OpenPecha/fast-antx.git",
+  "pandas<=2.2.3"
```
Comment (kaldan007) [inline pyproject.toml]: why did u remove that dependency. we need that dependency later for transfer

File: src/stt_data_with_llm/catalog_parser.py [hunk 2]
```diff
@@ -1,2 +1,73 @@
-def parse_catalog(catalog):
-    pass
\ No newline at end of file
+import logging
+
+import pandas as pd
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: if you are going to show the logging in terminal only how are you going to analysis the log later. I would save the log in a log file.

File: src/stt_data_with_llm/catalog_parser.py [hunk 3]
```diff
@@ -1,2 +1,73 @@
...
+            if not full_audio_id:
+                logging.warning(f"Row missing 'ID': {row}")
+                continue
+
+            audio_transcription_datas[full_audio_id] = {
+                "full_audio_id": full_audio_id,
+                "sr_no": row.get("Sr. no", ""),
+                "audio_url": row.get("Audio LInk", ""),
+                "reference_transcript": row.get("Audio text link", ""),
+                "speaker_id": row.get("Speaker ID", ""),
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: why are these attribute not parse
Speaker Name | Speaker Gender | News Channel | Publishing Year

File: src/stt_data_with_llm/catalog_parser.py [hunk 4]
```diff
@@ -1,2 +1,73 @@
...
+        try:
+            full_audio_id = row.get("ID", "")
+            if not full_audio_id:
+                logging.warning(f"Row missing 'ID': {row}")
+                continue
+
+            audio_transcription_datas[full_audio_id] = {
+                "full_audio_id": full_audio_id,
+                "sr_no": row.get("Sr. no", ""),
+                "audio_url": row.get("Audio LInk", ""),
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: incorrect key name if i refer to this [sheet](https://docs.google.com/spreadsheets/d/1H1QVQgux6C170GSKKJ10wzNh0I-o1OL4BlWqe2BfViA/edit?gid=436069356#gid=436069356):

**Review by kaldan007 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (jim-gyas) [reply pyproject.toml]: @kaldan007, I didn’t remove anything manually. After pulling the code, the changes to the dependencies didn’t reflect in my local setup. When I added the following dependency:
"git+https://github.com/OpenPecha/fast-antx.git"
It resulted in an error, stating that pyproject.toml does not allow Git-based dependencies to be specified in this format. To fix the issue, I changed it to:
"fast-antx @ git+https://github.com/OpenPecha/fast-antx.git"

File: src/stt_data_with_llm/catalog_parser.py (hunk 2 above)
Comment (jim-gyas) [reply src/stt_data_with_llm/catalog_parser.py]: @kaldan007, done with the changes.

File: src/stt_data_with_llm/catalog_parser.py (hunk 3 above)
Comment (jim-gyas) [reply src/stt_data_with_llm/catalog_parser.py]: @kaldan007, I was facing an issue with the news catalog earlier, as it wasn't working. The problem was caused by merging three rows to create the header, which led to the error. However, I have now resolved the issue, and the catalog is working fine.
### Error:
Screenshot 2024-12-25 at 11 03 04 AM

File: tests/data/actual_output/actual_audio_transcription_data.json
Comment (kaldan007) [inline tests/data/actual_output/actual_audio_transcription_data.json:1]: test data need not be that long.

Comment (kaldan007) [reply tests/data/actual_output/actual_audio_transcription_data.json:1]: it should rather cover corner cases

File: src/stt_data_with_llm/catalog_parser.py [hunk 5]
```diff
@@ -1,2 +1,101 @@
...
+    Returns:
+        pd.DataFrame: A cleaned DataFrame with rows and headers properly separated.
+    """
+    url = (
+        f"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv"  # noqa
+    )
+    try:
+        # Read the CSV data from the Google Spreadsheet
+        df = pd.read_csv(url, header=0)
+        print(df.head())
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: why this print is required.

File: src/stt_data_with_llm/catalog_parser.py [hunk 6]
```diff
@@ -1,2 +1,101 @@
...
+        # Log basic information about the DataFrame
+        logging.info("Spreadsheet successfully read.")
+        logging.info(f"Headers: {df.columns.tolist()}")
+        logging.info(f"First few rows:\n{df.head().to_string()}")  # noqa: E231
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: why are we logging first row

File: src/stt_data_with_llm/catalog_parser.py [hunk 7]
```diff
@@ -44,34 +44,51 @@ def catalog_parser(audio_url):
...
+                if not pd.isna(row.get("Publishing Year", ""))
+                else "",
             }
-            logging.info(f"print full audio id: %s" % full_audio_id)  # noqa: F541
+
         except Exception as e:
             logging.error(f"Error processing row: {row.to_dict()}. Error: {e}")
 
     logging.info(f"Parsed {len(audio_transcription_datas)} entries from the catalog.")
+    print("Data start")
```
Comment (kaldan007) [inline src/stt_data_with_llm/catalog_parser.py]: remove these lines

**Review by kaldan007 — APPROVED** [review]
